src/main/java/htsjdk/samtools/reference/IndexedFastaSequenceFile.java

```diff
@@ -111,178 +95,31 @@ public IndexedFastaSequenceFile(final Path path) throws FileNotFoundException {
         this(path, new FastaSequenceIndex((findRequiredFastaIndexFile(path))));
     }
 
-    @Override
-    public boolean isIndexed() {return true;}
-
-    private static File findFastaIndex(File fastaFile) {
-        File indexFile = getFastaIndexFileName(fastaFile);
-        if (!indexFile.exists()) return null;
-        return indexFile;
-    }
-
-    private static File getFastaIndexFileName(File fastaFile) {
-        return new File(fastaFile.getAbsolutePath() + ".fai");
-    }
-
-    private static File findRequiredFastaIndexFile(File fastaFile) throws FileNotFoundException {
-        File ret = findFastaIndex(fastaFile);
-        if (ret == null) throw new FileNotFoundException(getFastaIndexFileName(fastaFile) + " not found.");
-        return ret;
-    }
-
     public static boolean canCreateIndexedFastaReader(final File fastaFile) {
-        return (fastaFile.exists() &&
-                findFastaIndex(fastaFile) != null);
-    }
-
-    private static Path findFastaIndex(Path fastaFile) {
-        Path indexFile = ReferenceSequenceFileFactory.getFastaIndexFileName(fastaFile);
-        if (!Files.exists(indexFile)) return null;
-        return indexFile;
-    }
-
-    private static Path findRequiredFastaIndexFile(Path fastaFile) throws FileNotFoundException {
-        Path ret = findFastaIndex(fastaFile);
-        if (ret == null) throw new FileNotFoundException(ReferenceSequenceFileFactory.getFastaIndexFileName(fastaFile) + " not found.");
-        return ret;
+        return canCreateIndexedFastaReader(fastaFile.toPath());
     }
 
     public static boolean canCreateIndexedFastaReader(final Path fastaFile) {
```

Comment:
this is now unused, it should either be tested or deprecated.

Reply:
Deprecated in favor of `ReferenceSequenceFileFactory` implementation